Dalamud/Game/Chat/SeStringHandling/Payloads/AutoTranslatePayload.cs

```diff
@@ -1,105 +1,156 @@
+using Dalamud.Data.TransientSheet;
+using Lumina.Excel.GeneratedSheets;
+using Serilog;
 using System;
 using System.Collections.Generic;
 using System.IO;
-
+using System.Linq;
 
 namespace Dalamud.Game.Chat.SeStringHandling.Payloads
 {
-    public class AutoTranslatePayload : Payload
+    /// <summary>
+    /// An SeString Payload containing an auto-translation/completion chat message.
+    /// </summary>
+    public class AutoTranslatePayload : Payload, ITextProvider
     {
         public override PayloadType Type => PayloadType.AutoTranslateText;
 
-        public uint Group { get; set; }
+        private string text;
+        /// <summary>
+        /// The actual text displayed in-game for this payload.
+        /// </summary>
+        /// <remarks>
+        /// Value is evaluated lazily and cached.
+        /// </remarks>
+        public string Text
+        {
+            get
+            {
+                // wrap the text in the colored brackets that is uses in-game, since those
+                // are not actually part of any of the payloads
+                this.text ??= $"{(char)SeIconChar.AutoTranslateOpen} {Resolve()} {(char)SeIconChar.AutoTranslateClose}";
```

Comment:
Maybe don't wrap with the brackets by default?
The game probably does some handing itself of these payloads and wraps them after, similar to how it changes the text in emote text.

Reply:
The game just gets the binary payload, there is no text in it at all (or in any payloads that show up in chat when one of these is sent)

Reply:
The basic idea here is that .Text should be the actual text you see in the game, that you might expect to be in the payload (it just happens that for these payloads, they are entirely binary lookups)